File: demo/components/progressbar/progressbar-demo.ts
import {Component} from 'angular2/core';
import {CORE_DIRECTIVES} from 'angular2/common';
import {
  Ng2BootstrapConfig, Ng2BootstrapTheme, PROGRESSBAR_DIRECTIVES
} from '../../../ng2-bootstrap';

// switch bs3\bs4 templates

// webpack html imports
let templates:any = {
  [Ng2BootstrapTheme.BS3]: require('./progressbar-demo.html'),
  [Ng2BootstrapTheme.BS4]: require('./progressbar-demo-bs4.html')
};

@Component({
  selector: 'progressbar-demo',
  directives: [PROGRESSBAR_DIRECTIVES, CORE_DIRECTIVES],
  template: templates[Ng2BootstrapConfig.theme]
})
export class ProgressbarDemo {
  public max:number = 200;
  public showWarning:boolean;
  public dynamic:number;
  public type:string;
  public stacked:any[] = [];

  public constructor() {
    this.random();
    this.randomStacked();
  }

  public random():void {
    let value = Math.floor((Math.random() * 100) + 1);
    let type:string;

    if (value < 25) {
      type = 'success';
    } else if (value < 50) {
      type = 'info';
    } else if (value < 75) {
      type = 'warning';
    } else {
      type = 'danger';
    }

    this.showWarning = (type === 'danger' || type === 'warning');
    this.dynamic = value;
    this.type = type;
  };

  public randomStacked():void {
    let types = ['success', 'info', 'warning', 'danger'];

    this.stacked = [];
    let total = 0;
    for (let i = 0, n = Math.floor((Math.random() * 4) + 1); i < n; i++) {
      let index = Math.floor((Math.random() * 4));
      let value = Math.floor((Math.random() * 30) + 1);
      total += value;
      this.stacked.push({
        value: value,
        max: value, // i !== (n - 1) ? value : 100,
        type: types[index]
      });
    }
  };
}

File: web/components/progressbar/progressbar-demo.dart
import "package:angular2/angular2.dart";
import "package:ng2_strap/progressbar/progressbar.dart";
import 'dart:math';

@Component (selector: "progressbar-demo")
@View (templateUrl: "progressbar-demo.html",
    directives: const [ PROGRESSBAR_DIRECTIVES, CORE_DIRECTIVES, NgStyle])
class ProgressbarDemo {
  num max = 200;

  bool showWarning;

  num dynamic;

  String type;

  List stacked = [];

  ProgressbarDemo() {
    this.random();
    this.randomStacked();
  }

  random() {
    var value = new Random().nextInt(100);
    var type;
    if (value < 25) {
      type = "success";
    } else if (value < 50) {
      type = "info";
    } else if (value < 75) {
      type = "warning";
    } else {
      type = "danger";
    }
    this.showWarning =
    (identical(type, "danger") || identical(type, "warning"));
    this.dynamic = value;
    this.type = type;
  }

  randomStacked() {
    var types = [ "success", "info", "warning", "danger"];
    this.stacked = [];
    var total = 0;
    for (var i = 0, n = new Random().nextInt(5); i < n; i ++) {
      var index = new Random().nextInt(4);
      var value = new Random().nextInt(30);
      total += value;
      this.stacked.add({'value': value, 'max': value, 'type': types[index]});
    }
  }
}